### ISSUE TITLE
Persistence service no longer seems to work in OOP.

### ISSUE THREAD

**Issue by CyrusNajmabadi** [dotnet/roslyn#39653]
While debugging through https://github.com/dotnet/roslyn/issues/39364 i discovered something very bad.

Specifically, in OOP we don't get persistence working *at all* in some cases.  This is because operations like AddImport end up calling over to OOP which end updoing the search on a "RemoteTemporaryWorkspace".  I have no idea what thi workspace is, but it's *not* the actual RemoteWorkspace that OOP persistence was written to use.

This specifically breaks because we are unable to get a Storage instance back from teh storage service due to the final line here:

```c#
        internal IChecksummedPersistentStorage GetStorageWorker(Solution solution)
        {
            lock (_lock)
            {
                // Do we already have storage for this?
                if (solution.Id == _currentPersistentStorageSolutionId)
                {
                    // We do, great
                    return PersistentStorageReferenceCountedDisposableWrapper.AddReferenceCountToAndCreateWrapper(_currentPersistentStorage);
                }

                if (!SolutionSizeAboveThreshold(solution))
                {
                    return NoOpPersistentStorage.Instance;
                }

                var workingFolder = _locationService.TryGetStorageLocation(solution.Id);

                if (workingFolder == null)
                {
                    return NoOpPersistentStorage.Instance;
                }
```

There is no `workingFolder` because we have no actual `IPersistentStorageLocationService` for this workspace.  Specifically, we have defined IPersistentStorageLocationService for VS and for the normal RemoteWorkspace, like so:

```c#
    [ExportWorkspaceService(typeof(IPersistentStorageLocationService), layer: WorkspaceKind.RemoteWorkspace), Shared]
    internal class RemotePersistentStorageLocationService : IPersistentStorageLocationService
```

But we never defined this for the 'remote temporary workspace'.

Several things i don't understand at all here:

1. what is a "remote temporary workspace"?
2. why are we using it for some calls (like the call 'add import' makes to search the external index)?
3. what is even the right fix here? 

Basically, the entire scenario is broken making is so that that in OOP we have to compute the indices for at least some operations for the entire solutoin.  this likely leads to additional slowness and excess costs on startup for people trying to do normal operations that use OOP.

**Comment by CyrusNajmabadi** [comment]
tagging @mavasani @sharwell @heejaechang .  @heejaechang can you shed some light on what the 'remote temporary workspace' is for and how we would want to use it such that is can properly read/store to the persistence service?

Thanks!
